### PR TITLE
fix waiting for ressources

### PR DESCRIPTION
**Pull request by spickartz**
We need to tell the controller how many slots we want to allocate per
host.

### CODE REVIEW

File: src/scheduler_multi_app_consec.cpp [hunk 1]
```diff
@@ -23,7 +23,7 @@ void multi_app_sched_consec::schedule(const job_queueT &job_queue, fast::MQTT_co
 	// for all commands
 	for (auto job : job_queue.jobs) {
 		assert(job.req_cpus() <= controller.machines.size() * SLOT_SIZE * SLOTS);
-		controller.wait_for_ressource(job.req_cpus());
+		controller.wait_for_ressource(job.req_cpus(), 2);
```
Comment (jbreitbart) [inline src/scheduler_multi_app_consec.cpp]: Can you use the SLOTS constant here?

File: src/controller.cpp [hunk 2]
```diff
@@ -58,18 +58,20 @@ void controllerT::done() {
...
 		size_t counter = 0;
 
 		for (auto i : machine_usage) {
+			size_t allocated_slots = 0;
 			for (size_t s = 0; s < SLOTS; ++s) {
 				if (i[s] == std::numeric_limits<size_t>::max()) {
 					counter += SLOT_SIZE;
-					break;
+
+					if (allocated_slots++ == slots_per_host) break;
```
Comment (jbreitbart) [inline src/controller.cpp]: Hmm, that looks strange. Are you sure allocated_slots++ is correct?

File: src/scheduler_multi_app_consec.cpp (hunk 1 above)
Comment (spickartz) [reply src/scheduler_multi_app_consec.cpp]: I have changed that.

File: src/controller.cpp (hunk 2 above)
Comment (spickartz) [reply src/controller.cpp]: I guess you are right. Has been fixed.

**Pull request merged**